File: pinyougou-cart-web/src/main/resources/static/service/cartService.js
//购物车服务层
app.service('cartService', function($http){
    // 购物车列表
    this.findCartList = function(){
        return $http.get('/consumer/cart/findCartList');
    };

    // 添加商品到购物车
    this.addGoodsToCartList = function(itemId, num){
        return $http.get('/consumer/cart/' + itemId + '/' + num);
    };

    // 求合计
    this.sum = function(cartList){
        var totalValue = {totalNum:0, totalMoney:0.00 }; //合计实体
        for (var i = 0; i < cartList.length; i++) {
            var orderItemList = cartList[i].orderItemList;
            for (var j = 0; j < orderItemList.length; j++) {
                totalValue.totalNum += orderItemList[j].num;
                totalValue.totalMoney += orderItemList[j].totalFee;
            }
        }
        return totalValue;
    }
});

File: pinyougou-cart-interface/src/main/java/com/xt/pinyougou/service/CartService.java
package com.xt.pinyougou.service;

import com.xt.pinyougou.pojo.Cart;

import java.util.List;

/**
 * 购物车服务接口
 * @author Administrator
 *
 */
public interface CartService {


    /**
     * 添加商品到购物车
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num );

}
